### PR TITLE
Refactor Repeater model tracking

### PR DESCRIPTION
**Pull request by ogoffart**
Don't use weak-table

### CODE REVIEW

**Comment by ogoffart** [comment]
Note: the reason behind the change in the interpreter is that we only support inserting a ModelPeer in a single ModelNotify now.

File: sixtyfps_runtime/corelib/model.rs [hunk 1]
```diff
@@ -92,15 +122,13 @@ impl ModelNotify {
 impl ModelTracker for ModelNotify {
     /// Attach one peer. The peer will be notified when the model changes
     fn attach_peer(&self, peer: ModelPeer) {
-        peer.inner.upgrade().map(|rc| self.inner.borrow_mut().insert(rc));
+        if let Some(peer) = peer.inner.upgrade() {
+            self.inner().project_ref().repeaters.append(peer.as_ref())
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:126]: The terminology of ModelPeer and their attachment to a Model make me think that perhaps `repeaters` should be called `peers`.

File: sixtyfps_runtime/corelib/model.rs [hunk 2]
```diff
@@ -47,37 +57,57 @@ impl ModelTracker for () {
     fn track_row_count_changes(&self) {}
 }
 
+#[pin_project]
+#[derive(Default)]
+struct ModelNotifyInner {
+    #[pin]
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:63]: Nice to see pin_project in use :)

File: sixtyfps_runtime/corelib/model.rs (hunk 1 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/model.rs:126]: I wonder if it's worth to debug_assert here that the peer is not in the node list already?

**Review by tronical — APPROVED** [review]

File: sixtyfps_runtime/corelib/model.rs (hunk 1 above)
Comment (ogoffart) [reply sixtyfps_runtime/corelib/model.rs:126]: it might be in the list when we change model.

true, peers might be a better name

**Pull request merged**